examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/Logging.cs

```diff
@@ -0,0 +1,159 @@
+            generalDf = generalDf.Filter(generalDf["GeneralReg(value, genfilter)"] == true);
+            generalDf.Show();
+
+            // View the resulting schema
+            generalDf.PrintSchema();
+
+            // Step 2: Choose valid log entries that start with 10
+            spark.Udf().Register<string, string, bool>(
+                "IPReg",
+                (log, type) => RegTest(log, type));
+
+            generalDf.CreateOrReplaceTempView("IPLogs");
+
+            // Apply UDF to get valid log entries start with 10
+            DataFrame ipDf = spark.Sql(
+                "SELECT iplogs.value, IPReg(iplogs.value, 'ipfilter') FROM IPLogs");
+
+            ipDf.Show();
+
+            // Only keep log entries that matched both reg ex
+            ipDf = ipDf.Filter(ipDf["IPReg(value, ipfilter)"] == true);
+            ipDf.Show();
+
+            // Step 3: Choose valid log entries that start 
+            // with 10 and deal with spam
+            spark.Udf().Register<string, string, bool>(
+                "SpamRegEx",
+                (log, type) => RegTest(log, type));
+
+            ipDf.CreateOrReplaceTempView("SpamLogs");
+
+            // Apply UDF to get valid, start with 10, spam entries
+            DataFrame spamDF = spark.Sql(
+                "SELECT spamlogs.value, SpamRegEx(spamlogs.value, 'spamfilter') FROM SpamLogs");
+
+            spamDF.Show();
+
+            // Only keep log entries that matched all 3 reg ex
+            DataFrame trueSpam = spamDF.Filter(spamDF["SpamRegEx(value, spamfilter)"] == true);
+
+            // Formatting cleanup
+            // Use SQL to select just the entries, not boolean about reg ex
+            trueSpam.CreateOrReplaceTempView("TrueSpamLogs");
+            DataFrame trueSpamSql = spark.Sql(
+                "SELECT truespamlogs.value FROM truespamlogs");
+
+            // Explore the columns in the data we have filtered
+            // Let's try getting the number of GET requests
+            IEnumerable<Row> rows = trueSpamSql.Collect();
+            int numGetRequests = 0;
+            foreach (Row row in rows)
+            {
+                string rowstring = row.ToString();
+                numGetRequests += ParseLog(rowstring);
+            }
+
+            Console.WriteLine("Number of GET requests: " + numGetRequests);
+
+            spark.Stop();
+        }
+
+        public static bool RegTest(string logLine, string regexType)
+        {
+            // Which regex to use based on what we're filtering for
+            if (regexType == "ipfilter")
+            {
+                return Regex.IsMatch(logLine, "^(?=10)");
+            }
+            else if (regexType == "spamfilter")
+            {
+                return Regex.IsMatch(logLine, "\\b(?=spam)\\b");
+            }
+            else
+            {
+                return Regex.IsMatch(logLine, s_apacheRx);
+            }
+        }
+
+        public static int ParseLog(string logLine)
```

Comment:
Since this is only checking if the log has a GET, should we change the method name and what it returns reflect that better ?
```suggestion
        public static bool ContainsGet(string logLine)
```